native_client/modelstate.cc

```diff
@@ -65,11 +65,46 @@ ModelState::decode_metadata(const DecoderState& state,
     memcpy(&transcripts[i], &transcript, sizeof(CandidateTranscript));
   }
 
+  unsigned int num_timesteps = out[0].logits.size();
+  unsigned int alphabet_size = alphabet_.GetSize();
+
   Metadata* ret = (Metadata*)malloc(sizeof(Metadata));
-  Metadata metadata {
-    transcripts,  // transcripts
-    num_returned, // num_transcripts
-  };
-  memcpy(ret, &metadata, sizeof(Metadata));
-  return ret;
+
+  if (num_timesteps > 0) { // see if the logit structure has been filled and returned
+
+    AcousticModelEmissions* emissions = (AcousticModelEmissions*)malloc(sizeof(AcousticModelEmissions));
+
+    emissions->num_symbols = alphabet_size;
+    emissions->num_timesteps = num_timesteps;
+    emissions->symbols = (char**)malloc(sizeof(char*)*alphabet_size);
+    for(int i = 0; i < alphabet_size; i++) {
+        emissions->symbols[i] = strdup(alphabet_.DecodeSingle(i).c_str());
+    }
+
+    double* logits = (double*)malloc(sizeof(double)*alphabet_size*num_timesteps);
```

Comment:
I've changed it in this place, but should I change it across the board (e.g. in `struct Output` too)?

Reply:
Only in code you're adding.

Reply:
I think I added it in `struct Output`, and do you want it in the arguments / variables too? e.g. `keep_probs` instead of `keep_logits` and `WithProbs` instead of `WithLogits` ?

Reply:
Better yet `WithEmissions` to match the struct returned.

Reply:
Ok done.